### ISSUE TITLE
Normalize user's search text

### ISSUE THREAD

**Issue by ibleaman** [CloudCannon/pagefind#212]
I'm interested in normalizing the user's input text when doing a search.

For example, in the language I'm working with, the user might type in `ײ` (a single Unicode character) and I want to normalize it to `יי` (two Unicode characters) before doing the search, because my indexed pages use the two-character version. This is somewhat analogous in English to making text lowercase, or in other languages to stripping away _àccênt_ symbols, if the nuance is not important for search.

I've been able to achieve this by using this code on my search page:
```
<div id="search"></div>
<script>
    let pagefind = new PagefindUI({ element: "#search" });
    $("#search").on("keyup", function() {
      normalizedInput = normalizeUnicode(document.getElementsByTagName("input")[0].value);
      pagefind.triggerSearch(normalizedInput);
    });
</script>
```
where I've defined the function `normalizeUnicode(text)` elsewhere to make the necessary string replacements. However, this actually overwrites what's in the input box itself. I want to leave the user's input text as is (visually), but search using the replacements I've defined. (For example, if a user searched `áccent`, I want to search for `accent` but still leave the form `áccent` in the input box.) Is this possible? Thanks!

**Comment by bglw** [comment]
This is a good call! 

Ideally, normalizing between `áccent` and `accent` will be done automatically by Pagefind — currently just a few things around punctuation and casing are normalized. 

I haven't yet determined the best place to perform this logic, since I would like it to be bidirectional like the current normalization. A search for `accent` that returns `áccent` will need to have logic implemented in the Rust portion of the codebase, so it's a little harder to hook custom logic in for that direction. (i.e. in this case having `יי` return results for `ײ` — the search _result_ needs to be normalized, not the search term)

As a step in the right direction I could add an option to PagefindUI similar to [processResult](https://pagefind.app/docs/ui/#process-result), something like:

```js
new PagefindUI({
    element: "#search",
    processTerm: function (term) {
        return normalizeUnicode(term);
    }
});
```

That wouldn't handle the bidirectional aspect, but it sounds like it might still fit your use-case. Thoughts?

**Comment by ibleaman** [comment]
Your approach with `processTerm` could work very well for me! I agree that there is value in performing the same kind of text normalization on the search queries and the documents, but in my case the documents are already pretty much standardized, and so I just want to make some tweaks to the user's search term.

If `processTerm` would allow me to normalize the search term and perform a search, but **not** show the user what manipulations I've made to their term, that is great for me. Thank you!

(P.S. your video on pagefind is excellent and inspired me to use the tool on my jekyll site.)

**Comment by bglw** [comment]
Awesome to hear! I'll look at implementing this very soon 🙂

**Comment by bglw** [comment]
Hi @ibleaman — this is now released in [v0.11.0](https://github.com/CloudCannon/pagefind/releases/tag/v0.11.0) :)
